File: java/singleton/Singleton.java
public class Singleton {

    private static Singleton instance = null;

    private Singleton() {
    }

    // get instance of the object
    public static Singleton getInstance() {
        if (instance == null) {
            synchronized (Singleton.class) {
                if (instance == null) {
                    instance = new Singleton();
                }
            }
        }

        return instance;
    }

    // method
    public void showMessage() {
        System.out.println("Hello from the singleton instance ");
    }

}

File: typescript/singleton/src/singleton.ts
export class Singleton {
  private static instance: Singleton;

  private constructor() {}

  static getInstance(): Singleton {
    if (!this.instance) {
      Singleton.instance = new Singleton();
    }
    return Singleton.instance;
  }

  public someBusinessLogic() {
    console.log("Some business logic");
  }
}
